bpf/lib/conntrack.h

```diff
@@ -267,6 +267,33 @@ static __always_inline __u8 __ct_lookup(const void *map, struct __ctx_buff *ctx,
 	return CT_NEW;
 }
 
+/* The function determines whether an egress flow identified by the given
+ * tuple is a reply.
+ *
+ * The datapath creates a CT entry in a reverse order. E.g., if a pod sends a
+ * request to outside, the CT entry stored in the BPF map will be TUPLE_F_IN:
+ * pod => outside. So, we can leverage this fact to determine whether the given
+ * flow is a reply.
+ */
+#define CT_IS_REPLY(FAMILY)						\
+static __always_inline int						\
+ct_is_reply ## FAMILY(const void *map, struct __ctx_buff *ctx, int off,	\
```

Comment:
Ha, neat! I think more functions in the conntrack.h might benefit from this trick.

Maybe s/CT_IS_REPLY/DEFINE_FUNC_CT_IS_REPLY/ to make it more clear what the macro does.

Reply:
To be fair, I didn't invent this trick. We're already doing that a lot for the proxy and ipcache logic.